edgexfoundry/edgex-go
# feat(command): publish device service response to external MQTT

fix #4076 







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

1. Run core-command from this branch with external MQTT broker and `MESSAGEQUEUE_REQUIRED=true` environment variable
2. Run device-simple from device-sdk-go
3. Send command request to `edgex/command/request/<device-name>/<command-name>/<method>` topic. For example send following request to `edgex/command/Simple-Device01/Xrotation/get`:
```json
{
  "CorrelationID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
  "ApiVersion": "v2",
  "RequestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
  "ContentType": "application/json",
  "QueryParams": {
    "ds-pushevent": "no",
    "ds-returnevent": "yes"
  },
}

```
4. check the response is received on `edgex/command/response/<device-name>/<command-name>/<method>` topic:
```
{"ReceivedTopic":"edgex/command/response/device-simple/Simple-Device01/Xrotation/get","CorrelationID":"14a42ea6-c394-41c3-8bcd-a29b9f5e6835","ApiVersion":"v2","RequestID":"e6e8a2f4-eb14-4649-9e2b-175247911369","ErrorCode":0,"Payload":"eyJhcGlWZXJzaW9uIjoidjIiLCJzdGF0dXNDb2RlIjoyMDAsImV2ZW50Ijp7ImFwaVZlcnNpb24iOiJ2MiIsImlkIjoiZDVhMDA2YjYtYmY4OS00ZDQ4LWEyOWUtMTFhYzRlMzJjZWY3IiwiZGV2aWNlTmFtZSI6IlNpbXBsZS1EZXZpY2UwMSIsInByb2ZpbGVOYW1lIjoiU2ltcGxlLURldmljZSIsInNvdXJjZU5hbWUiOiJYcm90YXRpb24iLCJvcmlnaW4iOjE2NjM5MDgzMDQ1Mjk4OTEwMDAsInJlYWRpbmdzIjpbeyJpZCI6IjgwMTExNWI3LWEyZTEtNDE2Ny1iNzVmLWRkMjg1ODY2YjZlNiIsIm9yaWdpbiI6MTY2MzkwODMwNDUyOTg5MTAwMCwiZGV2aWNlTmFtZSI6IlNpbXBsZS1EZXZpY2UwMSIsInJlc291cmNlTmFtZSI6Ilhyb3RhdGlvbiIsInByb2ZpbGVOYW1lIjoiU2ltcGxlLURldmljZSIsInZhbHVlVHlwZSI6IkludDMyIiwidW5pdHMiOiJycG0iLCJ2YWx1ZSI6IjIwIn1dfX0=","ContentType":"application/json","QueryParams":{}}
```

## New Dependency Instructions (If applicable)
